widget/completionentry.go

```diff
@@ -162,6 +162,9 @@ func newNavigableList(items []string, entry *widget.Entry, setTextFromMenu func(
 				fn(i, o)
 				return
 			}
+			if i < 0 || i >= widget.ListItemID(len(n.items)) {
```

Comment:
Thanks for this fix, but I do wonder - should we guard the `fn` call above as well? It won't crash immediately but we may be leading the callback to crash or get unexpected callback data...

Reply:
What do you think @sharki13 could you move it so the `fn` callback does not get invalid data as well?

Reply:
Can this questions be answered @sharki13 ? It does seem like a good thing to have fixed.